### PR TITLE
Adds an Examine to Fix not knowing the cooldown of 8ball

### LINKED ISSUE #25822: The magic eightball has no progress bar when shaking
I thought that do_after was supposed to give a progress bar...

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
This is a fix unless @coiax wants to chime in and say it was intended to not know the time before you could use it again.

Also fixes #25822

### CODE REVIEW

**Comment by coiax** [comment]
While you're here, just change it to a sleep, rather than a do_after, the progress bar isn't appearing, and making them stand still for 15 seconds is silly.

**Comment by ExcessiveUseOfCobblestone** [comment]
literally just change do_after to sleep?

**Comment by coiax** [comment]
Well you'd need to refactor it slightly, and check that it hadn't been QDELETED in the sleep.

**Comment by ExcessiveUseOfCobblestone** [comment]
UGH!

**Comment by ExcessiveUseOfCobblestone** [comment]
@coiax ur a nerd

File: code/game/objects/items/eightball.dm [hunk 1]
```diff
@@ -41,6 +41,13 @@
 		new /obj/item/toy/eightball/haunted(get_turf(src))
 		qdel(src)
 
+/obj/item/toy/eightball/examine(mob/user)
+	..()
+	if(on_cooldown)
```
Comment (lzimann) [inline code/game/objects/items/eightball.dm]: I feel like it should display a message on use and not examine about the cooldown, it's better and more intuitive than on examine.

**Review by lzimann — CHANGES_REQUESTED** [review]

File: code/game/objects/items/eightball.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/eightball.dm]: It already does that, it just says you can't use it yet. Actually I'll fix it in a better way 1 sec

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/eightball.dm]: Hmmm.... @lzimann how might I go about telling it to print the time left [see line 69 I just don't know how to convert that into a number I can print]

Comment (lzimann) [inline code/game/objects/items/eightball.dm]: You don't need the time left, just saying that you can't use is enough(and it already does that).

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/eightball.dm]: yes but I /want/ the time left so I'm not shaking it constantly looking like a DUMMY!

Comment (coiax) [inline code/game/objects/items/eightball.dm]: Ask @MrStonedOne if it's possible to ask the timer subsystem how long it is until a timer fires

Comment (RemieRichards) [inline code/game/objects/items/eightball.dm]: @coiax it has to be possible, how do you think the timing subsystem does it? it's similar to qdelling too:
1. Add a timer set for 30 seconds in the future
2. Label that time as happening at ```world.time + 300ds```
3. Continually loop the list every process to see if any are ready, and if they are fire them off.

Even if it's not immediately possible, all it'd take is adding that future time onto the timer callback itself (it'd have to be a callback datum), and you'd have to find that callback datum by storing the timer in a variable (to get its ID)

tl;dr yes.

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/eightball.dm]: UGH OK I WHEN I CAN UNDERSTAND LADY REMIE AND HER WISDOM I WILL FIX THIS PROPERLY
